### DISCUSSION TITLE
[v5] Choosing the redirection url when validator fails

### DISCUSSION THREAD

**Discussion by ivan-prats** [adonisjs/core#2305] [Help] [answered]
Hello!

Is there any way of choosing which page to redirect-to in [v5] ("@adonisjs/core": "^5.0.4-preview-rc-2.1") when a validator fails?

```typescript
public async update({ request, response, params, session }: HttpContextContract) {
   // I want to be able to choose here where to redirect if it fails:
    const validatedData = await request.validate({ schema: commonMarketingProfileSchema })
    ....
}
```

My assumption is that it tells the browser to re-load the actual location (not sure tho). But I would like to force a redirection to another SomeController.edit route. In my dream-case scenario: I would love to choose where to redirect depending on the content-type request header.

My use-case is pretty niche and long to explain, plus I don't think it's going to help for this question ( it involves @hotwired/turbo lazy-loaded turbo-frames). But if needed: I can explain it.

Is this doable? I've seen in the old forum a `fails()` function that could be useful, but I haven't found anything in the docs nor in the type-autocompletion.

**Comment by ammezie** [comment, marked as answer]
One way you could go about that is to handle the validation fail exception inside the global exception handler and perform the redirect there.

Or:
https://preview.adonisjs.com/guides/validator/usage#standalone-usage

**Reply by ivan-prats** [reply]
Oh wow you are completely right: the only thing the validator does when it fails is throw an exception. The only thing I had to do was intercept it and do whathever I want with it.

Thanks @ammezie !
